### ISSUE TITLE
implement Docker API v1.41.1

### ISSUE THREAD

**Issue by mavogel** [kreuzwerker/terraform-provider-docker#142]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

- make all properties consistent (description (see #193 ) etc.) 
- be consistent with API [v1.41.1](https://docs.docker.com/engine/api/v1.41/)
- keep #34 and #85 in mind

### New or Affected Resource(s)

* docker_*

### References

* #94
* #34

**Comment by suzuki-shunsuke** [comment]
Please let me confirm what to do.

> * make all properties consistent (description etc.)

Does this mean to update field description in document and schema description in code according to the API document?

https://github.com/kreuzwerker/terraform-provider-docker/blob/6b0f977735dbc2ff99faa9278f4d51f2bf077b02/internal/provider/resource_docker_container.go#L247

> * be consistent with API v1.41.1

Does this mean to upgrade [Go Module of Docker API](https://github.com/kreuzwerker/terraform-provider-docker/blob/6b0f977735dbc2ff99faa9278f4d51f2bf077b02/go.mod#L8) or to support additional resources?

**Comment by mavogel** [comment]
> Does this mean to update field description in document and schema description in code according to the API document?

Yes and also. add missing properties of the API, which are not reflected yet in the schema. 

> Does this mean to upgrade Go Module of Docker API or to support additional resources?

IMHO we're already using the latest version since #119

**Comment by mavogel** [comment]
Related to #185

**Comment by tomalok** [comment]
I was thinking about this recently -- perhaps it's possible to generate the terraform schema directly from the official docker API definition?

**Comment by mavogel** [comment]
That's a very good point. Any idea/tools in mind @tomalok how to achieve this?

**Comment by tomalok** [comment]
I did a cursory search and found https://github.com/dikhan/terraform-provider-openapi which might be a good starting point / reference -- there may be others out there.

Obviously [Docker's swagger](https://github.com/docker/engine/blob/master/api/swagger.yaml) is a lot bigger than the examples...

**Comment by matifali** [comment]
are runtime arguments supported now?

**Comment by YurNaybor** [comment]
Sorry for digging out an old ticket, but is it now possible to configure capabilities for a service? I can't find it in the docs and trying to add  

```
      capabilities {
        drop = ["ALL"]
        add = ["CAP_CHOWN", "CAP_FOWNER"]
      }
```
to docker_service -> task_spec -> container_spec yields an error in provider version 3.0.2 "Blocks of type "capabilities" are not expected here."

**Comment by dhilgarth** [comment]
@mavogel : Has this ever been implemented?

**Comment by chrisclayson** [comment]
Am exceptionally interested in this as require the `cap_add` functionality for Swarm services (#34) ... whilst this issue was closed as completed, the requirement from #34 doesn't seem to have been included ... is there any roadmap for this? Currently, finding myself needing manually provision a service that requires specific capabilities 😞
